src/abstractmcmc.jl

```diff
@@ -303,3 +237,88 @@ function (cb::HMCProgressCallback)(rng, model, spl, t, state, i; nadapts = 0, kw
         @info "Finished $nadapts adapation steps" adaptor κ.τ.integrator metric
     end
 end
+
+#############
+### Utils ###
+#############
+
+function make_integrator(
+    rng::Random.AbstractRNG,
+    spl::Union{HMC,NUTS,HMCDA},
+    hamiltonian::Hamiltonian,
+    init_params,
+)
+    if iszero(spl.init_ϵ)
+        ϵ = find_good_stepsize(rng, hamiltonian, init_params)
+        @info string("Found initial step size ", ϵ)
+    end
+    integrator = eval(spl.integrator_method)
+    return integrator(ϵ)
+end
+
+function make_integrator(
+    rng::Random.AbstractRNG,
+    spl::HMCSampler,
+    hamiltonian::Hamiltonian,
+    init_params,
+)
+    # rerturns a dummy integrator
+    return Leapfrog(0.0)
```

Comment:
It shouldn't because it is not used anywhere when a `HMCSampler `is provided.
I have made it return a `AbstractIntegrator `now which should be more general.

Reply:
I found a more elegant solution!
I now simply returns the integrator provided by the user, even though it is technically not used anywhere.